### ISSUE TITLE
GPL-956 report on how many plates have been releases to postman pat [Heron]

### ISSUE THREAD

**Issue by rl15** [sanger/event_warehouse#160]
**User story**
As a product lead (Sara S) I would like a report on how many plates have been releases to postman pat on a given date

**Who are the primary contacts for this story**
Sara S (Product lead ops)
Rich C
James M

**Acceptance criteria**
To be considered successful the solution must allow:

- [ ] search a given date and see how many, and the tube IDs, of the pools released to postman pat. 
- [ ]  link those to the original 384 stamped plate ID (RT-Q barcode) also.

**Dependencies**

**Additional context**

**Comment by rl15** [comment]
James G wrote (Wednesday, 31 March 2021 at 15:53)

here’s a query to find all tubes scanned into Postman pat over March:
 
SELECT DISTINCT * FROM flat_events_view
LEFT OUTER JOIN metadata AS location ON location.event_id = flat_events_view.wh_event_id AND location.key = 'location_barcode'
WHERE
event_type = 'labware.received'
AND location.value = 'postmanpat'
AND role_type = 'labware'
AND occured_at >= '2021-03-00 00:00:00'
AND occured_at < '2021-04-00 00:00:00'
ORDER BY flat_events_view.wh_event_id DESC

**Comment by stevieing** [comment]
Conversation with James G (20/04). There is a great deal of difficulty getting information about the RT-Q plate. Apparently it would be much easier to get the LHTR cherrypick plate. Will put this on hold to get further clarity.

**Comment by stevieing** [comment]
Raw SQL from James G for above not using view:

SELECT
	          events.id AS wh_event_id,
	          events.uuid AS event_uuid_bin,
	          INSERT(INSERT(INSERT(INSERT(LOWER(HEX(events.uuid)),9,0,'-'),14,0,'-'),19,0,'-'),24,0,'-') AS event_uuid,
	          event_types.key AS event_type,
	          events.occured_at AS occured_at,
	          events.user_identifier AS user_identifier,
	          role_types.key AS role_type,
	          subject_types.key AS subject_type,
	          subjects.friendly_name AS subject_friendly_name,
	            INSERT(INSERT(INSERT(INSERT(LOWER(HEX(subjects.uuid)),9,0,'-'),14,0,'-'),19,0,'-'),24,0,'-') AS subject_uuid,
	          subjects.uuid AS subject_uuid_bin
	        FROM events
	        LEFT OUTER JOIN event_types ON events.event_type_id = event_types.id
	        LEFT OUTER JOIN roles ON roles.event_id = events.id
	        LEFT OUTER JOIN role_types ON roles.role_type_id = role_types.id
	        LEFT OUTER JOIN subjects ON roles.subject_id = subjects.id
	        LEFT OUTER JOIN subject_types ON subjects.subject_type_id = subject_types.id
	        LEFT OUTER JOIN metadata AS location ON location.event_id = events.id AND location.key = 'location_barcode'
	        WHERE
	        event_types.key = 'labware.received'
	        AND location.value = 'postmanpat' 
	        AND role_types.key = 'labware'
	        AND events.occured_at >= '2021-03-00 00:00:00'
	        AND events.occured_at < '2021-04-00 00:00:00'

**Comment by TWJW-SANGER** [comment]
The above should work but will be easier once DPL-773 is complete

**Comment by SujitDey2022** [comment]
@TWJW-SANGER can we close and archive this item as DPL-773 is completed?
